tests/v2/test_waku_peer_exchange.nim

```diff
@@ -126,8 +126,12 @@ procSuite "Waku Peer Exchange":
     await allFutures([node1.start(), node2.start(), node3.start()])
     await allFutures([node1.startDiscv5(), node2.startDiscv5()])
 
-    # Give disv5 some time to discover each other
-    await sleepAsync(5000.millis)
+    var attempts = 10
+    while (node1.wakuDiscv5.protocol.nodesDiscovered < 1 or
+          node2.wakuDiscv5.protocol.nodesDiscovered < 1) and
+          attempts > 0:
+      await sleepAsync(1.seconds)
+      attempts -= 1
```

Comment:
yup agree, the thing is that it might be tricky to have a generic proc, since in some cases it can be one condition, in others multiple ones (like here). each condition(s) can have a different value etc.

also i prefer an explicit check value1 == value2 instead of a proc just returning true/false and then asserting for that, since when asserting for value1==value2, you get the actual values, which gives more granularity for debugging.

but for sure, once we start doing this for other tests, i think we can come up with something more fancy than this.